utils/src/server/rpc_module.rs

```diff
@@ -252,6 +260,51 @@ impl SubscriptionSink {
 		for entry in errored {
 			subs.remove(&entry);
 		}
+
 		Ok(())
 	}
+
+	/// Send a message to all subscriptions that could parse `P` as input.
+	///
+	/// F: is a closure that you need to provide to apply on the input P.
+	pub fn send_each<T, F>(&self, f: F) -> anyhow::Result<()>
+	where
+		F: Fn(&mut Params) -> T,
+		T: Serialize,
+	{
+		let mut subs = self.subscribers.lock();
+		let mut errored = Vec::new();
+
+		for ((conn_id, sub_id), sender) in subs.iter_mut() {
+			let result = match sender.params.as_mut().map(|p| to_raw_value(&f(p))) {
+				Some(Ok(res)) => res,
+				_ => continue,
```

Comment:
I'm still unsure if / why we need to do this filtering here. The subscription method should either require params, or not. If the user wants to do optional params, they can do `register_subscription::<Option<T>>`.